docs: enrich migration guide (#1)
catenax-ng/tx-tractusx-edc: docs/migration/Version_0.1.x_0.3.x.md

# Migration from 0.1.x to 0.3.x

> ## Deprecation

Reviewer: ```suggestion

## Deprecation
```

- Removed field `assetId` from `ContractOffer`. It was always null though, so there should be nothing to do about it.
- on `POST /contractdefinitions` a `duration` field can be added to control the duration of the contract.
- added the `GET /assets/{id}/address` endpoint to being able to retrieve the stored `DataAddress`

## Settings changes

> - refactored the HTTP server contexts (more details on the [related decision record](https://eclipse-edc.github.io/docs/#/submodule/Connector/docs/developer/decision-records/2022-11-09-api-refactoring/renaming)). They need to be refactored as:

Reviewer: ```suggestion

- refactored the HTTP server contexts (more details on the [related decision record](https://eclipse-edc.github.io/docs/#/submodule/Connector/docs/developer/decision-records/2022-11-09-api-refactoring/renaming)). They need to be refactored as:
```

  - `web.http.data` becomes `web.http.management`
  - `web.http.ids` becomes `web.http.protocol`
  - `web.http.validation`, `web.http.controlplane` and `web.http.dataplane` become `web.http.control`
- Healthcheck api now it's exposed under the `management` context.
- Removed default value for setting `edc.transfer.proxy.token.verifier.publickey.alias` so it must be valued accordingly
- made the state machine settings configurable so it will be possible to tune them accordingly. More details in the [related documentation entry](https://eclipse-edc.github.io/docs/#/submodule/Connector/docs/developer/performance-tuning).
- renamed `edc.receiver.http.endpoint` to `edc.receiver.http.dynamic.endpoint`
- renamed `edc.oauth.public.key.alias` setting to `edc.oauth.certificate.alias`

## Other changes:

> - Supported `/public` data plane endpoint without trailing slash, that can be eventually removed from the configuration

Reviewer: ```suggestion

- Supported `/public` dataplane endpoint without trailing slash, that can be eventually removed from the configuration
```
